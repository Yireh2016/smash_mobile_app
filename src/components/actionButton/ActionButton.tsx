import * as React from 'react';
import { StyleProp, ViewStyle, View, StyleSheet } from 'react-native';
import { H2 } from '../texts/Texts';
import {
  TouchableOpacity,
  ContainedTouchableProperties,
} from 'react-native-gesture-handler';

import useMyTheme from '../../hooks/useMyTheme';

import Plus from '../../assets/Plus';

interface ActionButtonProps {
  label: string;
  onPress?: () => void;
  props?: ContainedTouchableProperties;
  style?: StyleProp<ViewStyle>;
  ref?: any;
  disabled?: boolean;
}

const ActionButton: React.FunctionComponent<ActionButtonProps> = ({
  onPress,
  label,
  style,
  props,
  disabled = false,
}) => {
  const theme = useMyTheme();
  const styles = StyleSheet.create({
    btnContainer: {
      backgroundColor: disabled ? theme.palette.gray : theme.palette.tertiary,
      borderRadius: 12,
      width: '100%',
    },
    btnLabel: {
      color: disabled ? theme.palette.black : theme.palette.white,
      textAlign: 'left',
    },
    btnLayout: {
      display: 'flex',
      flexDirection: 'row',
      paddingLeft: 16,
      paddingRight: 16,
      paddingTop: 20,
      paddingBottom: 20,
    },
    btnLabelContainer: {
      flexGrow: 90,
    },
    btnPlusContainer: {
      flexGrow: 10,
      alignItems: 'flex-end',
    },
  });

  return (
    <View style={styles.btnContainer}>
      <TouchableOpacity
        testID="ActionButton"
        disabled={disabled}
        style={style}
        {...props}
        onPress={onPress}
      >
        <View style={styles.btnLayout}>
          <View style={styles.btnLabelContainer}>
            <H2
              color={
                disabled ? theme.palette.black : theme.palette.primaryContrast
              }
              style={styles.btnLabel}
            >
              {label}
            </H2>
          </View>
          <View style={styles.btnPlusContainer}>
            <Plus
              color={disabled ? theme.palette.white : theme.palette.secondary}
              isRound={true}
            />
          </View>
        </View>
      </TouchableOpacity>
    </View>
  );
};

export default ActionButton;
