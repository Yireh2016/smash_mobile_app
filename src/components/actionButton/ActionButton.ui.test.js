import * as React from 'react';
import renderer from 'react-test-renderer';
import ActionButton from './ActionButton';

it('ActionButton renders correctly', () => {
  const ActionButtonProps = {
    onPress: () => jest.mock(),
    label: 'Test',
  };
  const Element = () => {
    return <ActionButton {...ActionButtonProps} />;
  };

  const actionBtn = renderer.create(<Element />).toJSON();
  expect(actionBtn).toMatchSnapshot();
});
