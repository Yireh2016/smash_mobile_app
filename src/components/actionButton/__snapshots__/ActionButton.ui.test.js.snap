// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2C3D49",
      "borderRadius": 12,
      "width": "100%",
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="ActionButton"
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "paddingBottom": 20,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexGrow": 90,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 20,
                "textAlign": "left",
              }
            }
          >
            Test
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flexGrow": 10,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={25}
            bbWidth={25}
            focusable={false}
            height={25}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 25,
                  "width": 25,
                },
              ]
            }
            vbHeight={25}
            vbWidth={25}
            width={25}
            xml="
    <svg xmlns=\\"http://www.w3.org/2000/svg\\" width=25 height=25 viewBox=\\"0 0 25 25\\">
  <path id=\\"plus\\" d=\\"M12.5,0A12.5,12.5,0,1,0,25,12.5,12.514,12.514,0,0,0,12.5,0Zm5.469,13.542H13.542v4.427a1.042,1.042,0,0,1-2.083,0V13.542H7.031a1.042,1.042,0,0,1,0-2.083h4.427V7.031a1.042,1.042,0,0,1,2.083,0v4.427h4.427a1.042,1.042,0,0,1,0,2.083Zm0,0\\" fill=#29b782/>
</svg>

    "
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M12.5,0A12.5,12.5,0,1,0,25,12.5,12.514,12.514,0,0,0,12.5,0Zm5.469,13.542H13.542v4.427a1.042,1.042,0,0,1-2.083,0V13.542H7.031a1.042,1.042,0,0,1,0-2.083h4.427V7.031a1.042,1.042,0,0,1,2.083,0v4.427h4.427a1.042,1.042,0,0,1,0,2.083Zm0,0"
                fill={4280924034}
                name="plus"
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
